### PR TITLE
Test manager sends AR log format as expected

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|---|
|[#2285](https://github.com/wazuh/wazuh-qa/issues/2285) - https://github.com/wazuh/wazuh/issues/10858 |

## Description
The test checks that when the manager sends the Active Response messages they are escaped in the expected format, according to the Agent version, to avoid code injection. 
The test works by inserting a log into an Agent, and the expected Log appearing in `active-responses.log` and/or `ossec.log` must show the expected format:

|Agent Version | Expected Format |
|--|--|
| version < 4.2.0 | A series of characters are escaped with a **\** |
| between 4.2.0 and 4.2.4 | **$**, **&**, **`** and **'** are escaped with hexadecimal notation |
| versions 4.2.5 and up | The inserted log appears with no escapes - the issue is fixed |

## Configuration options
In order to run the test, first the environment located in `/test/system/provisioning/manager_agent` must be enabled with: 
`sudo ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch": "4.2"}'`

## Tests
|Run| Results | Date | By | Notes |
|--|--|--|--|--|
| R1 | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7719758/R1.zip)  | 2021/12/15 | @Deblintrake09 | Results show a warning about local cache, but that is not related to the test itself but on how my machine was configured.
| R2 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7719759/R2.zip)  | 2021/12/15 | @Deblintrake09  | Results show a warning about local cache, but that is not related to the test itself but on how my machine was configured.
| R3 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7719760/R3.zip) | 2021/12/15 | @Deblintrake09  | Results show a warning about local cache, but that is not related to the test itself but on how my machine was configured.
| R1-fixed | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7734900/R1-fixed.zip) | 2021/12/17 | @Deblintrake09  | Results show a warning about local cache, but that is not related to the test itself but on how my machine was configured.
| R4 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755878/R4.zip) | 2021/12/21 | @Deblintrake09  | Fixed - Now test cleans environment after running. Randomized log IP
| R5 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755875/R5.zip) | 2021/12/21 | @Deblintrake09  | 
| R6 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755876/R6.zip) | 2021/12/21 | @Deblintrake09  | 
| R7 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755877/R7.zip) | 2021/12/21 | @Deblintrake09  | 
| R8 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755880/R8.zip) | 2021/12/21 | @Deblintrake09  | 
| R9 | [ :green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755879/R9.zip) | 2021/12/21 | @Deblintrake09  | Multiple instances - to show it works properly

- [x] Proven that tests **pass** when they have to pass.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] The test is documented in wazuh-qa/docs.

### CODE REVIEW

File: tests/system/provisioning/manager_agent/README.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
+# wazuh-qa
+
+Wazuh - Basic cluster provisioning
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/system/provisioning/manager_agent/README.md:3]: This file was not modified. Will apply corrections to whole file.

File: deps/wazuh_testing/wazuh_testing/tools/monitoring.py [hunk 2]
```diff
@@ -834,7 +834,8 @@ def wrapper(*args, **kwargs):
 
 def callback_generator(regex):
     def new_callback(line):
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/tools/monitoring.py:836]: renamed to generate_monitoring_callback.
Added description.

File: deps/wazuh_testing/wazuh_testing/tools/system.py [hunk 3]
```diff
@@ -63,6 +63,13 @@ def add_block_to_file(self, host: str, path: str, replace: str, before: str, aft
                                     check=check)
 
     def modify_file_content(self, host: str, path: str = None, content: str = ''):
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/tools/system.py:65]: This method even though it works with files, it is related to working on a remote host. If we move it over to `file.py`, we would need to move everything related to hosts over for the method to work.

**Comment by CamiRomero** [comment]
## Tests
|Run| Results | Date | By | Notes |
|--|--|--|--|--|
| R1 | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755729/R1.zip)  | 21/12/2021  | @CamiRomero  | |
| R2 | [:red_circle:](https://github.com/wazuh/wazuh-qa/files/7755730/R2.zip)  | 21/12/2021 | @CamiRomero  | The test failed after re run without re-created the environmnet. @Deblintrake09  will work on it|
| R3 | [:red_circle:](https://github.com/wazuh/wazuh-qa/files/7755731/R3.zip) | 21/12/2021 |  @CamiRomero | Same as befor|

**Comment by CamiRomero** [comment]
## Tests
|Run| Results | Date | By | Notes |
|--|--|--|--|--|
| R1 | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/7755991/R1.zip) | 21/12/2021  | @CamiRomero  | |
| R2 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7755997/R2.zip) | 21/12/2021 | @CamiRomero  ||
| R3 | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/7755998/R3.zip)| 21/12/2021 |  @CamiRomero | |

**Review by damarisg — APPROVED** [review]

**Pull request merged**